servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -424,9 +432,20 @@ protected AbstractDnsSubscription newSubscription(
             return new AbstractDnsSubscription(subscriber) {
                 @Override
                 protected Future<DnsAnswer<HostAndPort>> doDnsQuery(final boolean scheduledQuery) {
-                    Promise<DnsAnswer<HostAndPort>> promise = nettyIoExecutor.eventLoopGroup().next().newPromise();
+                    final EventLoop eventLoop = nettyIoExecutor.eventLoopGroup().next();
+                    final Promise<DnsAnswer<HostAndPort>> promise = eventLoop.newPromise();
+                    final Future<?> timeoutFuture = resolutionTimeoutMillis == 0L ? null : eventLoop.schedule(() -> {
+                        if (promise.isDone()) {
+                            return;
+                        }
+                        promise.tryFailure(DnsNameResolverTimeoutException.newInstance(
+                                name, resolutionTimeoutMillis, SRV.toString(), SrvRecordPublisher.class, "doDnsQuery"));
```

Comment:
wdyt about cancelling the underlying resolution if we succeed in failing `promise`? It looks like it won't be properly acted upon by netty, but otoh it's good practice in case it ever is honored.

Reply:
good idea, please check that I didn't mess up with anything: https://github.com/apple/servicetalk/pull/2908/commits/c86221fab6c80d5dc2890c357700e72d0fc98b59